### DISCUSSION TITLE
Questions about secondary development

### DISCUSSION THREAD

**Discussion by ghyaauoo** [langgenius/dify#6476] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Provide a description of requested docs changes

Is it possible to perform secondary development based on a specific version of Dify, integrating the system into our own platform, and allowing single sign-on to Dify? We do not intend to offer Dify as a SaaS product externally; it will be used solely as a process design tool and to leverage its related API capabilities. Will there be any copyright issues associated with this?

**Comment by crazywoola** [comment]
See our license https://docs.dify.ai/policies/open-source

Dify is permitted to be used for commercialization, such as using Dify as a "backend-as-a-service" for your other applications, or delivering it to enterprises as an application development platform. However, when the following conditions are met, you must contact the producer to obtain a commercial license:

Multi-tenant SaaS service: Unless explicitly authorized by Dify in writing, you may not use the Dify.AI source code to operate a multi-tenant SaaS service that is similar to the Dify.AI service edition.

LOGO and copyright information: In the process of using Dify, you may not remove or modify the LOGO or copyright information in the Dify console.

If you do not meet one of them or all of them, you should contact business@dify.ai.

**Comment by stanyang0001** [comment]
Hello, I would like to know if we need to consider the LOGO and copyright information when we integrate them into our own system by using API publishing mode. Do we have to keep these information for display in our own system?

Thanks.

**Reply by crazywoola** [reply]
If you are using something like `https://api.dify.ai/v1` which you deployed by yourself. Then, no. You do not need to concern about the logo stuff.

**Reply by stanyang0001** [reply]
Thank you for your kind help. I got it.
Great product!

**Reply by LydiaDaddy** [reply]
@crazywoola  “What it means is that as long as we download the source code and develop it further based on that, we won’t encounter any commercial copyright issues, right?”

**Reply by crazywoola** [reply]
> @crazywoola “What it means is that as long as we download the source code and develop it further based on that, we won’t encounter any commercial copyright issues, right?”

I don't think so, see apache license and our open source license. If you need to change the Logo or enable multiple tetant, you need to obtain a business license from us.

**Comment by monsterindian** [comment]
I would like to white label this and sell with my branding. I would also like to make changes and features to this. All my clients are financial institutions hence I'll never have One instance used by multiple clients. Is rebranding allowed?
